lib/libthr/thread/thr_stack.c

```diff
@@ -294,23 +288,30 @@ _thr_stack_alloc(struct pthread_attr *attr)
 		/* Release the lock before mmap'ing it. */
 		THREAD_LIST_UNLOCK(curthread);
 
+#ifdef __CHERI_PURE_CAPABILITY__
+		/*
+		 * Grouping stacks together is not useful on CHERIABI, we let
+		 * let mmap() decide where to allocate
```

Comment:
I kind of wish there was a way to request a search from the top of the address space down so stacks were at the top as a matter of developer ergonomics.